service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -0,0 +1,118 @@
+public class DraftCMOServiceTest {
+
+    @Autowired
+    private DraftCMOService draftCMOService;
+
+    private final LocalDate date = LocalDate.now();
+
+    @Test
+    void getHearingDatesDynamic_shouldReturnHearingDateDynamicList() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date))
+            )
+            .build();
+
+        DynamicList hearingList = draftCMOService.getHearingDatesDynamic(caseDetails);
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+
+    }
+
+    @Test
+    void getHearingDatesDynamicShouldFillTheHearingDatesListWithPreviousSelectedValue() {
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date),
+                "caseManagementOrder", createCaseManagementOrder(draftCMOService.convertDate(date.plusDays(2)))
+                )).build();
+
+        DynamicList hearingList = draftCMOService.getHearingDatesDynamic(caseDetails);
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+    }
+
+    @Test
+    void shouldReturnHearingDateDynamicListIfHearingDatesNotNull() {
+        DynamicList hearingList = draftCMOService.buildDynamicListFromHearingDetails(
+            createHearingBookings(date));
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+    }
+
+    @Test
+    void convertDateShouldReturnDateInUKFormat() {
+        String dateInUKFormat = draftCMOService.convertDate(LocalDate.of(2019,11,11));
+        assertThat(dateInUKFormat).isEqualTo("11 Nov 2019");
+    }
+
+    private CaseManagementOrder createCaseManagementOrder(String hearingDate) {
+        return CaseManagementOrder.builder()
+            .hearingDate(hearingDate)
+            .hearingDateId(UUID.fromString("6b3ee98f-acff-4b64-bb00-cc3db02a24b2").toString()).build();
```

Comment:
Do we need to to build UUID.fromString()? Use `UUID.randomUUID()`

Reply:
UUID.fromString ... to string... why 😂